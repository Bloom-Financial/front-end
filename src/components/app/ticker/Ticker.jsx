import React from 'react';
import PropTypes from 'prop-types';

const Ticker = ({ ticker, name, market, locale, type, currency, active, primaryExch, url }) => (

  
  <figure>
    <figcaption>
      <p>{ticker}</p>
      <p>{name}</p>
      <p>{market}</p>
      <p>{locale}</p>
      <p>{type}</p>
      <p>{currency}</p>
      <p>{active}</p>
      <p>{primaryExch}</p>
      <p>{url}</p>
      
    </figcaption>
  
  </figure>
);

Ticker.propTypes = {
  ticker: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  market: PropTypes.string.isRequired,
  local: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  primaryExch: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Ticker;

