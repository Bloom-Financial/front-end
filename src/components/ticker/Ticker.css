.Ticker {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  font-family: 'Fjalla One', sans-serif;
  color: seashell;
  letter-spacing: 1px;
  padding: 5px;
}