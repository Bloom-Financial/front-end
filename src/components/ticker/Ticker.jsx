import React from 'react';
import PropTypes from 'prop-types';

function Ticker({ symbol, regularMarketPrice, previousClose }) {
  console.log(symbol, 'slb')
  return (
    <>
      {symbol}
      {previousClose}
      {regularMarketPrice}
    </>
  );
}

// Ticker.propTypes = {
//   quotes: PropTypes.array.isRequired
// };

export default Ticker;




