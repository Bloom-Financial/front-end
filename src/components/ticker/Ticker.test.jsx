import React from 'react';
import { render } from '@testing-library/react';
import Ticker from './Ticker';

describe('Test the ticker component', () => {
  it('tests the ticker component', () => {
    const { asFragment } = render(<Ticker/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
