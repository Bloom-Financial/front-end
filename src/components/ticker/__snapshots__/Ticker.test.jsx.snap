// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the ticker component tests the ticker component 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
